src/navigation/breadcrumbs/breadcrumbs.jsx

```diff
@@ -143,49 +173,52 @@ function Breadcrumbs(props) {
     const pathName = get(router, 'location.pathname');
     const pathParams = get(router, 'params');
     const prevPathName = useRef();
+    const prevRoutesNameMapper = useRef();
     const routerPushFunction = get(router, 'push');
 
     useEffect(() => {
-        const routes = get(router, 'routes', []);
-        const currentExistentRoutes = routesToArray(routes);
-        setExistentRoutes(currentExistentRoutes);
-        const currentBreadCrumbs = getPathNameBreadcrumbs(
-            pathName,
-            pathParams,
-            routerPushFunction,
-            currentExistentRoutes,
-        );
-        setPathBreadcrumbs(currentBreadCrumbs);
+        if (!staticCrumbs.length > 0 && router) {
+            const routerRoutes = get(router, 'routes', []);
+            const currentExistentRoutes = routesToArray(routerRoutes);
+            setExistentRoutes(currentExistentRoutes);
+            const currentBreadCrumbs = getPathNameBreadcrumbs(
+                pathName,
+                pathParams,
+                routerPushFunction,
+                currentExistentRoutes,
+                routesNameMapper,
+            );
+            setPathBreadcrumbs(currentBreadCrumbs);
+        }
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []); // didMount
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
     useEffect(() => {
-        const shouldUpdatePathBreadcrumbs = prevPathName.current &&
-            prevPathName.current !== pathName;
+        const shouldUpdatePathBreadcrumbs = !staticCrumbs.length > 0 && router &&
```

Comment:
`!staticCrumbs.length > 0` looks not quite right.  What is the intent here?

Reply:
changed to `staticCrumbs.length === 0`.

Reply:
Yeah, `staticCrumbs.length === 0` seems better than `!staticCrumbs.length > 0`.  I'll resolve this.